plenum/server/replica.py

```diff
@@ -1049,11 +1057,17 @@ def process_three_phase_msg(self, msg: ThreePhaseMsg, sender: str):
         """
         sender = self.generateName(sender, self.instId)
 
+        pp_key = (msg, sender) if isinstance(msg, PrePrepare) else None
+        if pp_key and (pp_key not in self.pre_prepare_tss):
+            self.pre_prepare_tss[pp_key] = self.utc_epoch
```

Comment:
I think it's better to use `self.get_time_for_3pc_batch` for better unit tests

Reply:
done